v2/jdbc-to-googlecloud/src/main/java/com/google/cloud/teleport/v2/templates/JdbcToBigQuery.java

```diff
@@ -160,21 +165,53 @@ static PipelineResult run(JdbcToBigQueryOptions options, Write<TableRow> writeTo
     PCollection<TableRow> rows;
     if (options.getPartitionColumn() != null && options.getTable() != null) {
       // Read with Partitions
-      // TODO(pranavbhandari): Support readWithPartitions for other data types.
-      JdbcIO.ReadWithPartitions<TableRow, Long> readIO =
-          JdbcIO.<TableRow>readWithPartitions()
+      JdbcIO.ReadWithPartitions<TableRow, ?> readIO = null;
+      final String partitionColumntype = options.getPartitionColumnType();
+      if(partitionColumntype == null ){ 
+        throw new IllegalStateException("Expected non-null partitionColumnType");
```

Comment:
this will make change backward incompatible. Imagine users currently running it with partitionColumn that is of long type and everything was working. Now it won't work. We should keep column type default to null unless configured differently

Reply:
agree, or assume null as long , to make it work like before